### ISSUE TITLE
`.descendants()` failed with master-part tables where the descendants only contain the part-table

### ISSUE THREAD

**Issue by ttngu207** [datajoint/datajoint-python#927]
### Description
`table.descendants()` failed for the case where there are master-part tables downstream, where only the part-table is a descendant, but not the master.

### Reproducibility
Include:
- OS (WIN)
- Python 3.7
- MySQL Version 5.7
- MySQL Deployment Strategy (local-native)
- DataJoint Version 0.13.2

#### Reproducible case

Given the following schema design

```python
import datajoint as dj
schema = dj.Schema('test')

@schema
class A(dj.Manual):
    definition = """
    a: int
    """


@schema
class B(dj.Manual):
    definition = """
    -> A
    b: int
    """


@schema
class Master(dj.Manual):
    definition = """
    table_master: int
    """

    class Part(dj.Part):
        definition = """
        -> master
        -> B
        """
```

calling `A.descendants()` will result in the error below:

```
Traceback (most recent call last):
  File "C:\Users\thinh\Documents\TN-Vathes\MAP\map-ephys\venv\lib\site-packages\IPython\core\interactiveshell.py", line 3343, in run_code
    exec(code_obj, self.user_global_ns, self.user_ns)
  File "<ipython-input-3-a68a3ab7b68c>", line 1, in <module>
    TableA.descendants()
  File "C:\Users\thinh\Documents\TN-Vathes\MAP\map-ephys\venv\lib\site-packages\datajoint\table.py", line 181, in descendants
    for node in self.connection.dependencies.descendants(self.full_table_name) if not node.isdigit()]
  File "C:\Users\thinh\Documents\TN-Vathes\MAP\map-ephys\venv\lib\site-packages\datajoint\dependencies.py", line 149, in descendants
    nx.algorithms.dag.topological_sort(nodes)))
  File "C:\Users\thinh\Documents\TN-Vathes\MAP\map-ephys\venv\lib\site-packages\datajoint\dependencies.py", line 30, in unite_master_parts
    raise DataJointError("Found a part table {name} without its master table.".format(name=name))
datajoint.errors.DataJointError: Found a part table `test`.`master_table__part_table` without its master table.
```

**Comment by guzman-raphael** [comment]
@ttngu207 Thanks for the report and it is unfortunate to hear of your issue. Have you by chance confirmed if this is a regression issue or not? As in, does this work as expected in `0.12.x`?

**Comment by ttngu207** [comment]
This is likely a logic issue, I've confirmed that this issue is also in `0.12.x`

**Comment by chrisroat** [comment]
I also have run into this in the context of doing a topological sort.  In that example, there is a flow that involves  PreprocessDone+PreprocessDone.Part as a master-part table:

Preprocess -> PreprocessDone.Part 
PreprocessDone -> Analyze

[This test](https://github.com/chrisroat/dj_pipeline/blob/d5799318e81391ad3d469129adae37fdf2d87984/test_pipeline.py#L22) shows the resultant topological sort, where Analyze comes before Preprocess.

I use the sorted table order to push data through our cluster.  I just need to run it a few times to make sure things all complete.  :)

**Comment by chrisroat** [comment]
Thanks @guzman-raphael .  Will a fix for this qualify for 0.13.3?
